src/commands/ext-diagnose.ts

```diff
@@ -0,0 +1,20 @@
+import { checkMinRequiredVersion } from "../checkMinRequiredVersion";
+import { Command } from "../command";
+import { needProjectId } from "../projectUtils";
+import { diagnose } from "../extensions/diagnose";
+import { ensureExtensionsApiEnabled } from "../extensions/extensionsHelper";
+import { requirePermissions } from "../requirePermissions";
+
+module.exports = new Command("ext:diagnose")
+  .description("diagnoses the Firebase project for potential known configuration issues")
+  .option("--fix", "automatically perform basic fixes")
+  .before(requirePermissions, [
+    "resourcemanager.projects.getIamPolicy",
+    "resourcemanager.projects.setIamPolicy",
+  ])
+  .before(ensureExtensionsApiEnabled)
+  .before(checkMinRequiredVersion, "extMinVersion")
+  .action((options: any) => {
```

Comment:
Use the Options type instead to silence all these annoying warnings: https://github.com/firebase/firebase-tools/blob/master/src/options.ts. You'll just need to add the new flag to that type

Reply:
Done.